nsls2forge_utils/cli.py

```diff
@@ -49,5 +50,45 @@ def main():
         check_package_version(**version_kwargs)
 
 
-if __name__ == '__main__':
-    sys.exit(main())
+def all_feedstocks():
+    parser = argparse.ArgumentParser(
+        description='Retrieve all feedstock repositories from cache or GitHub '
+                    'and write to file.')
+
+    # Set file path
+    parser.add_argument('-f', '--filepath', dest='filepath',
+                        default='names.txt', type=str,
+                        help=('filepath to write feedstock names to '
+                              '(default is names.txt)'))
+
+    # Set cached to true
+    parser.add_argument('-c', '--cached', dest='cached',
+                        action='store_true',
+                        help=('read the names of feedstocks from the cache'))
+
+    # Give GitHub organization name
+    parser.add_argument('-o', '--organization', dest='organization',
+                        default=None, type=str,
+                        help=('GitHub organization to get feedstocks from '
+                              '(must be specified if not cached)'))
+
+    # Give GitHub username and token
+    parser.add_argument('-u', '--username', dest='username',
+                        default=None, type=str,
+                        help=('GitHub username for authentication'))
+    parser.add_argument('-t', '--token', dest='token',
+                        default=None, type=str,
+                        help=('GitHub token for authentication'))
+
+    args = parser.parse_args()
+
+    if not args.cached and args.organization is None:
+        parser.print_help()
+
+    names = get_all_feedstocks(cached=args.cached,
+                               organization=args.organization,
+                               username=args.username,
+                               token=args.token)
+    names = sorted(names)
+    if not args.cached:
+        _write_list_to_file(names, args.filepath)
```

Comment:
Looks like we do nothing if it's cached. I think we should print the names unconditionally, and have a separate CLI argument to write them to file. I don't think we should couple the caching flag with writing.